clients/src/main/java/org/apache/kafka/clients/admin/ConsumerGroupDescription.java

```diff
@@ -17,55 +17,96 @@
 
 package org.apache.kafka.clients.admin;
 
+import org.apache.kafka.common.ConsumerGroupState;
+import org.apache.kafka.common.Node;
 import org.apache.kafka.common.utils.Utils;
 
-import java.util.List;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Objects;
 
 /**
  * A detailed description of a single consumer group in the cluster.
  */
 public class ConsumerGroupDescription {
-
     private final String groupId;
     private final boolean isSimpleConsumerGroup;
-    private final List<MemberDescription> members;
+    private final Collection<MemberDescription> members;
     private final String partitionAssignor;
+    private final ConsumerGroupState state;
+    private final Node coordinator;
 
-    /**
-     * Creates an instance with the specified parameters.
-     *
-     * @param groupId               The consumer group id
-     * @param isSimpleConsumerGroup If Consumer Group is simple
-     * @param members               The consumer group members
-     * @param partitionAssignor     The consumer group partition assignor
-     */
-    public ConsumerGroupDescription(String groupId, boolean isSimpleConsumerGroup, List<MemberDescription> members, String partitionAssignor) {
-        this.groupId = groupId;
+    public static class Builder {
```

Comment:
> Curious if we use builders for any of the other admin object?

I would argue the *Options classes (CreateTopicsOptions, etc.) are very similar to builders.  For example, you can use CreateTopicsOptions like this:  `new CreateTopicsOptions().timeoutMs(10000).validateOnly(true)`  This is useful to let the options evolve over time.

I think `ConsumerGroupDescription` will almost certainly get new fields over time, so some of the same concerns apply here (although it's less urgent if the constructor is not public)

> Does it need to be public?

I guess we can make it package-private for now.  Code that needs this for unit tests could use  `MockAdminClient`.

Reply:
Yeah, builders are nice for the reasons you mention. However, since it wasn't discussed in KIP-222, I think keeping it package private for now might be better.